multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -101,7 +121,20 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 			return ctrl.Result{}, err
 		}
 		// If MemberClusterAnnounce is deleted, no need to process because RemoveMember must already
-		// be called.
+		// be called. Then the Reconciler will delete the member cluster in the ClusterSet's member list
+		// in the leader cluster.
+		memberClusterAnnounceObj, installed, _ := r.installedMemberClusterAnnounce.GetByKey(req.NamespacedName.String())
+		if !installed {
+			klog.InfoS("MemberClusterAnnounce is not installed in leader cluster, no need to delete member cluster in ClusterSet", "MemberClusterAnnounce", req.String())
```

Comment:
Probably remove "in leader cluster, no need to delete member cluster in ClusterSet".